content/_includes/layout.macros.njk

```diff
@@ -52,15 +52,15 @@ params:
 {%- endmacro %}
 
 {% macro _grid_content(data) %}
-  {% for item in data %}
-    {% set id = [item.title, loop.index] | join('-') | slug %}
-    <article class="grid-item" aria-labelledby="{{ id }}">
-      <h3 class="item-title" id="{{ id }}">
-        {{ item.title | mdInline | safe }}
-      </h3>
-      {{ item.text | md | safe }}
-    </article>
-  {% endfor %}
+{%- for item in data -%}
+  {%- set id = [item.title, loop.index] | join('-') | slug -%}
+  <article class="grid-item" aria-labelledby="{{ id }}">
+    <h3 class="item-title" id="{{ id }}">
+      {{ item.title | mdInline | safe }}
+    </h3>
+    {{ item.text | md | safe }}
+  </article>
+{%- endfor -%}
```

Comment:
Whoa. It makes me nervous that this minor whitespace change can cause that bug... I guess because we're parsing it as Markdown... 😬

Reply:
Yeah, this is the issue I warned about in my very first presentation of 11ty as an option. HTML is valid in markdown - and generally not re-parsed. But the problem is caused specifically by indentation that makes it look like part of a markdown code block.

Right now the safest approach to mixing `md` with `njk` is to set the file type (or `templateEngineOverride`) to `njk`, and then wrap markdown in a "paired shortcode" block:

```njk
{% md %}
# This is markdown in njk
{% endmd %}
```

We do that a few places – and could make it the standard if we want – but we're not consistent, and it often feels like overkill when you just want to include a couple macros in an article. I'd love to find a better long-term solution. Maybe a filter that strips out indentation? Seems like a basic html minification script would do it…?

Reply:
Ok, makes sense. The problem with a filter is remembering to use it -- similar to the current problem, no? Unless it's applied automatically/globally.

I'd love an approach that we don't have to remember to use -- that is, it either breaks (with an error we can see) or works safely.

Reply:
Yeah, that would be best. Make a story for it?